### ISSUE TITLE
tailscale ssh: cannot connect using Blink iOS ssh client

### ISSUE THREAD

**Issue by abeluck** [tailscale/tailscale#4997]
### What is the issue?

I saw on HN the invitation to report tailscale ssh bugs to this repo.

I’m using [Blink](https://github.com/blinksh/blink) on my iPad. 

I cannot connect from my iPad to a linux box running tailscale with `—ssh`

### Steps to reproduce

1. Setup tailscale ssh on a linux box
2. Install Blink and tailscale on an iPad
3. Try to ssh from the iPad to the linux box

```

blink> ssh -vvv gitpod@ 100.99.236.199
Setting up connection callbacks.
Starting connection to 100.99.236.199
ssh_connect: libssh 0.9.3 (c) 2003-2019 Aris Adamantiadis, Andreas Schneider and libssh contributors. Distributed under the LGPL, please refer to COPYING file for information about your rights, using threading threads_pthread
ssh_connect: Socket connecting, now waiting for the callbacks to work
ssh_connect: current state : 1
-[IO stream:handleEvent:]: Received POLLOUT in connecting state
socket_callback_connected: Socket connection callback: 1 (0)
Starting connection to 100.99.236.199
ssh_connect: current state : 2
callback_receive_banner: Received banner: SSH-2.0-Tailscale
ssh_client_connection_callback: SSH server banner: SSH-2.0-Tailscale
ssh_analyze_banner: Analyzing banner: SSH-2.0-Tailscale
ssh_client_select_hostkeys: Order of wanted host keys: "ssh-ed25519,ecdsa-sha2-nistp521,ecdsa-sha2-nistp384,ecdsa-sha2-nistp256,rsa-sha2-512,rsa-sha2-256,ssh-rsa,ssh-dss"
ssh_known_hosts_read_entries: Failed to open the known_hosts file '/etc/ssh/ssh_known_hosts': No such file or directory
ssh_client_select_hostkeys: Algorithms found in known_hosts files: "ssh-ed25519"
ssh_client_select_hostkeys: Changing host key method to "ssh-ed25519,ecdsa-sha2-nistp521,ecdsa-sha2-nistp384,ecdsa-sha2-nistp256,rsa-sha2-512,rsa-sha2-256,ssh-rsa,ssh-dss"
packet_send2: packet: wrote [type=20, len=1020, padding_size=10, comp=1009, payload=1009]
ssh_send_kex: SSH_MSG_KEXINIT sent
Starting connection to 100.99.236.199
ssh_connect: current state : 4
ssh_packet_socket_callback: packet: read type 20 [len=604,padding=5,comp=598,payload=598]
ssh_packet_process: Dispatching handler for packet type 20
ssh_kex_select_methods: Negotiated curve25519-sha256,ssh-ed25519,aes128-gcm@openssh.com,aes128-gcm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-256-etm@openssh.com,none,none,,
packet_send2: packet: wrote [type=30, len=44, padding_size=6, comp=37, payload=37]
Starting connection to 100.99.236.199
ssh_connect: current state : 6
ssh_packet_socket_callback: packet: read type 31 [len=188,padding=8,comp=179,payload=179]
ssh_packet_process: Dispatching handler for packet type 31
packet_send2: packet: wrote [type=21, len=12, padding_size=10, comp=1, payload=1]
crypt_set_algorithms2: Set output algorithm to aes128-gcm@openssh.com
crypt_set_algorithms2: Set HMAC output algorithm to aead-gcm
crypt_set_algorithms2: Set input algorithm to aes128-gcm@openssh.com
crypt_set_algorithms2: Set HMAC input algorithm to aead-gcm
ssh_init_rekey_state: Set rekey after 4294967296 blocks
ssh_init_rekey_state: Set rekey after 4294967296 blocks
ssh_packet_client_curve25519_reply: SSH_MSG_NEWKEYS sent
ssh_packet_socket_callback: Processing 16 bytes left in socket buffer
ssh_packet_socket_callback: packet: read type 21 [len=12,padding=10,comp=1,payload=1]
ssh_packet_process: Dispatching handler for packet type 21
ssh_packet_newkeys: Received SSH_MSG_NEWKEYS
ssh_packet_newkeys: Signature verified and valid
Starting connection to 100.99.236.199
ssh_connect: current state : 7
Connection succeeded...
Authenticating...
Trying none...
packet_send2: packet: wrote [type=5, len=32, padding_size=14, comp=17, payload=17]
ssh_service_request: Sent SSH_MSG_SERVICE_REQUEST (service ssh-userauth)
ssh_packet_socket_callback: packet: read type 6 [len=32,padding=14,comp=17,payload=17]
ssh_packet_process: Dispatching handler for packet type 6
ssh_packet_service_accept: Received SSH_MSG_SERVICE_ACCEPT
packet_send2: packet: wrote [type=50, len=48, padding_size=10, comp=37, payload=37]
ssh_packet_socket_callback: packet: read type 51 [len=32,padding=16,comp=15,payload=15]
ssh_packet_process: Dispatching handler for packet type 51
ssh_packet_userauth_failure: Access denied for 'none'. Authentication that can continue: tailscale
ssh_packet_userauth_failure: Access denied for 'none'. Authentication that can continue: tailscale
Error connecting to 100.99.236.199. authError(msg: "Could not authenticate, no valid methods to try.")
```

Relevant ACL:

```
"ssh": [
		// Allow all users to SSH into their own devices in check mode.
		// Comment this section out if you want to define specific restrictions.
		{
			"action": "check",
			"src":    ["autogroup:members"],
			"dst":    ["autogroup:self"],
			"users":  ["autogroup:nonroot", "root"],
		},
		{
			"action": "accept",
			"src":    ["autogroup:members"],
			"dst":    ["tag:myservertag"],
			"users":  ["root", "muuser@domain.com"],
		},
	],
```

### OS

iOS

### OS version

15.4.1

### Tailscale version

1.26.1

### Bug report

BUG-10430395aecc31777707996fd919e41fcc5ac2b288323b06a9957546bc0065d9-20220701211308Z-7b7e560f33fa0eb0

**Comment by DilumAluthge** [comment]
I am able to reproduce this.

My hypothesis is that this happens because check mode is enabled and you need to re-SSO. The bug here seems to be that Blink does not print the message from Tailscale that tells you that you need to re-SSO.

I am able to get SSH to work if I apply either of the following two workarounds:
1. In your ACL, change "check" to "accept". (This workaround is unfortunate because it disables check mode.)
2. On your iOS device, go to the Tailscale app, click the three dots in the top right hand corner to open the menu, and click on "Reauthenticate". Once you have reauthenticated, go to the Blink app and try to SSH.

**Comment by DentonGentry** [comment]
The iOS Tailscale app is supposed to show a local notification that authentication is needed in these cases, added because Blink does not have handling to do so. You don't see a notification from the Tailscale app?

**Comment by bradfitz** [comment]
I just use Blink SSH on my iPhone to Tailscale SSH in check mode and it worked.

I got the Tailscale notification at the top,

<img width="419" alt="Screen Shot 2022-07-01 at 2 53 52 PM" src="https://user-images.githubusercontent.com/2621/176971922-b3ef222f-a0fa-4706-b176-1270c41765f3.png">

... clicked it, went back to Blink and I was in.

**Comment by DilumAluthge** [comment]
I was not getting a notification, but that's because I had apparently disabled notifications for the Tailscale app.

I went to the iOS Settings app and re-enabled notifications for the Tailscale app, and now I do receive the notification from Tailscale when I try to SSH in Blink.

Perhaps it would be good to add a note to the Tailscale iOS documentation explaining that users should enable/allow notifications?

**Comment by bradfitz** [comment]
@mihaip made a change just the other day to set the "time sensitive" bit on those notifications,

```patch
         let content = UNMutableNotificationContent()
         ...
+        content.interruptionLevel = .timeSensitive
```

@DentonGentry, let's cherry-pick that change too (in the iOS repo) for 1.26.2.

**Comment by DilumAluthge** [comment]
I realize now that I misread the OP's log, and I think the issue I was encountering with Blink is different from the OP's issue.

In my case, when I opened Blink and did `user@host`, it would hang indefinitely while it waited for me to re-SSO.

However, in the OP's issue, they are getting an "Could not authenticate, no valid methods to try" error.

**Comment by abeluck** [comment]
Yea..

I can't get it to work at all. I have notifications enabled, and never received a notification. 

As my logs show, there is connectivity to the ssh host, but the ssh authentication negotiation is not working.

**Comment by DilumAluthge** [comment]
Are you able to Tailscale SSH into this Linux machine from a different client machine (other than your iPad)?

**Comment by abeluck** [comment]
No.. which led me to playing with the ACL rules. This is my first time using tailscale ACL rules, so I had it configured incorrectly

### Not working ACL
```
		{
			"action": "accept",
			"src":    ["autogroup:members"],
			"dst":    ["tag:gitpod"],
			"users":  ["root", "myuser@domain.com"],
		},
````

This acl doesn't work becuase the user in the gitpod container is `gitpod`

### Working ACL

```
		{
			"action": "accept",
			"src":    ["autogroup:members"],
			"dst":    ["tag:gitpod"],
			"users":  ["autogroup:nonroot"],
		},
```

This works because `"autogroup:nonroot"` allows logins to the `gitpod` user.

Sorry for the noise, clearly a case of RTFM.

I also changed `accept` to `check` and I did receive a notification.

**Comment by DentonGentry** [comment]
The `time-sensitive` bit will be in 1.26.2.

**Comment by DentonGentry** [comment]
I'm unclear on whether there is still a problem, or if everything is resolved by the fixes in Tailscale 1.26.2.

**Comment by bradfitz** [comment]
`ssh_packet_userauth_failure: Access denied for 'none'. Authentication that can continue: tailscale` means it was an ACL problem, which OP realized in https://github.com/tailscale/tailscale/issues/4997#issuecomment-1172864264 so it looks like there's nothing remaining to do on this bug.
